### PR TITLE
Changed the lambda func to async and returning promise

### PR DESCRIPTION
**Pull request by skscheduleonce**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -7,28 +7,40 @@ const apiVersion = process.env.APIVERSION;
...
+            })
+          });
+        }
+        resolve({
+          'statusCode': res.statusCode,
+          'body': JSON.stringify({
+            message: res,
+          })
+        });
+      });
```
Comment (mderazon) [inline index.js]: According to the superagent doc, this is not how you _await_ for a request
```suggestion
return request
  .post(
    `https://${workspaceId}.ods.opinsights.azure.com/api/logs?api-version=${apiVersion}`
  )
  .send(payload)
  .set(headers);
```
`end()` doesn't need to be used when making an async call, success/rejection should be handled in lambda in this case (need to test)

**Review by mderazon — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (skscheduleonce) [reply index.js]: Removed the end() now and also added try and catch block to handle the exception. Also tested on sam local cli and it is working fine.

Comment (mderazon) [reply index.js]: have you tried my suggestion ?

Comment (skscheduleonce) [reply index.js]: Yes, I have tried the same way. Please check the updated code.

**Review by mderazon — APPROVED** [review]

**Pull request merged**